File: java/SortedSquares.java
public class SortedSquares {
    

    public int[] sortedSquares(int[] nums){
        int n = nums.length;
        int left = 0; 
        int right = n-1;
        int[] res = new int[n];

        for(int i=n-1; i >=0 ; i-- ){
            int square = 0;
            if(Math.abs(nums[left]) > Math.abs(nums[right])){
                square = nums[left];
                left++;
            } else {
                square = nums[right];
                right--;
            }
            res[i] = square * square;
        }
        return res;
    }
}

File: python/arraysandstrings/sorted_squares.py
def sortedSquares(nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        n = len(nums)
        left = 0
        right = len(nums) -1
        res = [0] * n
        
        #  [-4,-1,0,3,10]
        
        for i in range(n-1, -1, -1 ):
            if(abs(nums[left])  > abs(nums[right]) ):
                square = nums[left]
                left+= 1
            else:
                square = nums[right]
                right -= 1
                
            res[i] = square * square
        return res

print(sortedSquares([-4,-1,0,3,10]))